src/SmartComponents/Forms/Helpers/schemaParser.js

```diff
@@ -0,0 +1,369 @@
+import setWith from 'lodash/setWith';
+
+/**Fields component placeholders */
+const TEXT_FIELD = 'text-field';
+const TEXTAREA_FIELD = 'textarea-field';
+const FIELD_ARRAY = 'field-array';
+const SELECT_COMPONENT = 'select-field';
+const FIXED_LIST = 'fixed-list';
+const CHECKBOX = 'checkbox-field';
+const SUB_FORM = 'sub-form';
+const RADIO = 'radio-field';
+
+/**Validator functions placeholder */
+const REQUIRED = 'required-validator';
+const MIN_LENGTH = 'min-length-validator';
+const MIN_ITEMS_VALIDATOR = 'min-items-validator';
+const MIN_NUMBER_VALUE = 'min-number-value';
+const MAX_NUMBER_VALUE = 'max-number-value';
+
+let autofocusField = undefined;
+let definitions = undefined;
+let defaultValues = {};
+
+const validatorBuilder = ({ schema, fields = {}, key }) => {
+    const result = [];
+    if (schema.required && schema.required.includes(key)) {
+        result.push({ type: REQUIRED });
+    }
+
+    if (fields[key] && fields[key].minLength) {
+        result.push({ type: MIN_LENGTH, treshold: fields[key].minLength });
+        delete fields[key].minLength;
+    }
+
+    if (schema.minItems) {
+        result.push({ type: MIN_ITEMS_VALIDATOR, treshold: schema.minItems });
+    }
+
+    return result;
+};
+
+const componentMapper = (type, dataType) => ({
+    string: { component: TEXT_FIELD, type: 'text', dataType },
+    color: { component: TEXT_FIELD, type: 'color', dataType },
+    hidden: { component: TEXT_FIELD, type: 'hidden', dataType },
+    tel: { component: TEXT_FIELD, type: 'tel', dataType },
+    email: { component: TEXT_FIELD, type: 'email', dataType },
+    password: { component: TEXT_FIELD, type: 'password', dataType },
+    integer: { component: TEXT_FIELD, type: 'number', step: 1, dataType },
+    updown: { component: TEXT_FIELD, type: 'number', dataType },
+    number: { component: TEXT_FIELD, type: 'number', dataType },
+    range: { component: TEXT_FIELD, type: 'range', dataType },
+    textarea: { component: TEXTAREA_FIELD, dataType },
+    select: { component: SELECT_COMPONENT, dataType },
+    boolean: { component: CHECKBOX, type: 'checkbox', dataType },
+    checkbox: { component: CHECKBOX, type: 'checkbox', dataType },
+    checkboxes: { component: CHECKBOX, type: 'checkbox', dataType },
+    radio: { component: RADIO, type: 'radio', dataType }
+})[type];
+
+const createFieldOptions = (options = {}) => {
+    const result = {};
+    const uiOption = options['ui:options'] || {};
+    if (options['ui:disabled']) {
```

Comment:
You are technically right from JS perspective. I was also thinking about persisting this format into some DB so i have tried to minimize the size of the output (that is also why i am deleting some keys that are not required or just duplicates). But i don't really know if it is relevant.

Reply:
> I guess we can remove the if statements and just use plain assign.

I think this won't work because there can be multiple options and they are scattered across different keys and nested structures. It can look something like this:

```javascript
{
    "ui:disabled": true,
    "ui:options": {
      "backgroundColor": "pink"
      ...
    }
    ...
}
```
The options is no flat structure (don't ask me why i think its stupid...)